cli/internal/llm/callOpenAI.go

```diff
@@ -0,0 +1,150 @@
+package llm
+
+import (
+	"context"
+	"fmt"
+	"hyaline/internal/config"
+	"log/slog"
+
+	"github.com/openai/openai-go/v2"
+	"github.com/openai/openai-go/v2/option"
+)
+
+func callOpenAI(systemPrompt string, userPrompt string, tools []*Tool, cfg *config.LLM) (result string, err error) {
+	if cfg.Key == "" {
+		slog.Warn("Calling OpenAI without a key being set")
+	}
+	var clientOptions []option.RequestOption
+	clientOptions = append(clientOptions, option.WithAPIKey(cfg.Key))
+
+	if cfg.Provider == config.LLMProviderGitHubModels {
+		if cfg.Endpoint == "" {
+			slog.Debug("Using default GitHub Models endpoint")
+			clientOptions = append(clientOptions, option.WithBaseURL("https://models.github.ai/inference"))
```

Comment:
Let's make sure to document this default in the config reference (that we default the endpoint to models... only for the github-models provider